ui/components/Header/Header.js

```diff
@@ -1,80 +1,125 @@
-              {pageTitle}
-            </Typography>
-          </Grid>
-
-          {/* <Grid item className={classes.notifications}>
-                <MesheryNotification />
-              </Grid> */}
-          <Grid item className={classes.userContainer}>
-            <IconButton color="inherit" size="large">
-              <Link href="/system/connections">
-                <img
-                  src={
-                    pageTitle === "Connection Wizard"
-                      ? "/static/img/connection_wizard/connection-wizard-green.svg"
-                      : "/static/img/connection_wizard/connection-wizard-white.svg"
-                  }
-                  className={
-                    classes.headerIcons + " " + (pageTitle === "Connection Wizard" ? classes.itemActiveItem : "")
-                  }
-                />
-              </Link>
-            </IconButton>
-
-            <IconButton color="inherit" size="large">
-              <Link href="/settings">
-                <SettingsIcon
-                  className={classes.headerIcons + " " + (pageTitle === "Settings" ? classes.itemActiveItem : "")}
-                />
-              </Link>
-            </IconButton>
-
-            {/* <div data-test="notification-button">
-              <MesheryNotification />
-            </div> */}
-            {/* <span className={classes.userSpan}>
-              <User color="inherit" iconButtonClassName={classes.iconButtonAvatar} avatarClassName={classes.avatar} />
-            </span> */}
-          </Grid>
-        </Grid>
-      </Toolbar>
-    </AppBar>
-  );
-};
-
-export default HeaderComponent;
+/* eslint-disable react/prop-types */
+import React from "react";
+import SettingsIcon from "@mui/icons-material/Settings";
+import { AppBar, Grid, Hidden, IconButton, Toolbar, Typography, useTheme } from "@mui/material";
+import { styled } from "@mui/material/styles";
+import Link from "next/link";
+import MenuIcon from "@mui/icons-material/Menu";
+import User from "../User";
+
+const UserSpan = styled(({ children, ...props }) => <span {...props}>{children}</span>)(({ theme }) => ({
+  marginLeft: theme.spacing(1),
+}));
+
+const UserGrid = styled(Grid)(() => ({
+  paddingLeft: 1,
+  display: "flex",
+  alignItems: "center",
+}));
+
+const HeaderIcon = styled(({ icon, isActive = false, children, ...props }) => {
+  const Component = icon;
+  return (
+    <Component
+      style={{
+        color: isActive ? "#00B39F" : "inherit",
+      }}
+      {...props}
+    >
+      {children}
+    </Component>
+  );
+})(() => ({
+  fontSize: "1.5rem",
+  height: "1.5rem",
+  width: "1.5rem",
+}));
+
+const PageTitleWrapperGrid = styled(Grid)(() => ({ flexGrow: 1, marginRight: "auto" }));
+
+const PageTitle = styled(Typography)(({ theme }) => ({
+  paddingLeft: theme.spacing(2),
+  fontSize: "1.25rem",
+  [theme.breakpoints.up("sm")]: { fontSize: "1.65rem" },
+}));
+
+const HeaderComponent = ({ drawerOpen, onDrawerToggle, pageTitle }) => {

```

Comment:
Where this drawerOpen coming from??
I was asking because the state of drawer should be a global redux variable. We are need of it around different components inside meshery-extensions too where prop drilling is just impossible.

Reply:
Yes, you are right, we need to have the state of drawer in redux. Currently it is passed from the Layout component. I think someone else is working on the drawer. @YashKamboj  could tell that. Should I work on moving the drawer state to redux and create a hook to access drawer state in my next PR?